infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -219,11 +222,28 @@ static ByteBuffer getByteBufferWithMetaData(AbstractMessage message) {
     }
 
     @Override
-    public TailsResponse getTails() {
+    public StreamsAddressResponse getStreamsAddressSpace() {
+        Map<UUID, StreamAddressSpace> streamsAddressSpace = new HashMap<>(logMetadata.getStreamsAddressSpaceMap());
+        return new StreamsAddressResponse(streamsAddressSpace);
+    }
+
+    @Override
+    public LogAddressSpaceResponse getLogAddressSpace() {
+        Map<UUID, StreamAddressSpace> streamsAddressSpace = new HashMap<>(logMetadata.getStreamsAddressSpaceMap());
```

Comment:
Copy an entire map - seems like a performance/memory impact

Reply:
This is actually not needed! Removed.